src/variorum/Intel_DGPU/power_features.c

```diff
@@ -0,0 +1,136 @@
+// Copyright 2019-2022 Lawrence Livermore National Security, LLC and other
+// Variorum Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: MIT
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <power_features.h>
+#include <config_architecture.h>
+#include <variorum_error.h>
+#include <variorum_timers.h>
+#include <libapmidg.h>
+
+static unsigned m_total_unit_devices;
+static unsigned m_gpus_per_socket;
+static char m_hostname[1024];
+
+void initAPMIDG(void)
+{
+    int verbose = 0;
+    unsigned m_num_package;
+    apmidg_init(verbose);
+
+    m_total_unit_devices = apmidg_getndevs();
+
+    /* Collect number of packages and GPUs per package */
+    variorum_get_topology(&m_num_package, NULL, NULL);
+    m_gpus_per_socket = m_total_unit_devices / m_num_package;
+
+    /* Save hostname */
+    gethostname(m_hostname, sizeof(m_hostname));
+}
+
+void shutdownAPMIDG(void)
+{
+    apmidg_finish();
+}
+
+void get_power_data(int chipid, int verbose, FILE *output)
+{
+    int d;
+    static int init_output = 0;
+
+    //Iterate over all GPU device handles for this socket and print power
+    for (d = chipid * (int)m_gpus_per_socket;
+         d < (chipid + 1) * (int)m_gpus_per_socket; ++d)
+    {
+        int pi = 0; // only report the global power domain now
+        double value = apmidg_readpoweravg(d, pi);
+
+        if (verbose)
+        {
+            fprintf(output,
+                    "_GPU_POWER_USAGE Host: %s, Socket: %d, DeviceID: %d, Power: %lf W\n",
```

Comment:
Can we prefix Intel here in the output like we do for most platforms: `GPU_POWER_USAGE` --> `INTEL_GPU_POWER_USAGE`. Same for temperature and clocks in other places